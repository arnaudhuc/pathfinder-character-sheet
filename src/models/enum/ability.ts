export enum EAbility {
	STR = 'STR',
	DEX = 'DEX',
	CON = 'CON',
	INT = 'INT',
	WIS = 'WIS',
	CHA = 'CHA',
	FRE = 'FRE',
	NUL = 'NUL',
}
