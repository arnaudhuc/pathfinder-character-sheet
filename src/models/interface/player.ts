export interface IPlayerState {
	characterName: string;
	playerName: string;
}
