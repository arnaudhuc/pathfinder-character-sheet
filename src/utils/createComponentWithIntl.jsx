import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import messages_en from '../i18n/en.json';

const createComponentWithIntl = (
	children,
	props = { locale: 'en', messages: messages_en },
) => {
	return renderer.create(<IntlProvider {...props}>{children}</IntlProvider>);
};
export default createComponentWithIntl;
