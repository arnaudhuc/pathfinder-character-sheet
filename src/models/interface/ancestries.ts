import { EAbility } from '../enum/ability';

export interface IAncestries {
	id: number;
	name: string;
	imageUrl: string;
	abilityBoost: Array<EAbility>;
	abilityFlaw: Array<EAbility>;
	isClicked: boolean;
}
