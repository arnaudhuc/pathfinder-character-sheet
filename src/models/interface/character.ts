import { EAbility } from '../enum/ability';

export interface ICharacterState {
	ancestry: string;
	strengthScore: number;
	dexterityScore: number;
	constitutionScore: number;
	intelligenceScore: number;
	wisdomScore: number;
	charismaScore: number;
	abilityBoost: Array<EAbility>;
	abilityFlaw: Array<EAbility>;
}
