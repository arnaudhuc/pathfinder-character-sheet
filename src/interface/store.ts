import { IPlayerState } from '../models/interface/player';

export interface IStore {
	characterReducer: IPlayerState;
}

export interface ICharacter {
	name: string;
}
