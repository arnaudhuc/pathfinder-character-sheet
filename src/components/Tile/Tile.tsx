/* eslint-disable react/prop-types */
/** @jsx jsx */
import { jsx } from '@emotion/core';
import { useIntl } from 'react-intl';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Typography from '@material-ui/core/Typography';
import { green } from '@material-ui/core/colors';
import CheckCircleIcon from '@material-ui/icons/CheckCircle';
import { Divider } from '@material-ui/core';

import { ITileInfo } from '../../models/interface/ancestries';

interface ITile {
	handleClick: (id: number) => void;
	ancestry: ITileInfo;
}

export const Tile: React.FC<ITile> = (props) => {
	const { ancestry, handleClick } = props;
	const { formatMessage } = useIntl();

	const useStyles = makeStyles({
		root: {
			maxWidth: 240,
			border: ancestry.isClicked
				? `2px solid ${green.A200}`
				: '1px solid rgba(0, 0, 0, 0.12)',
			position: 'relative',
			overflow: 'unset',
		},
	});

	const classes = useStyles();

	const cardUseStyles = makeStyles({
		root: {
			height: ancestry.isClicked ? 270 : 250,
			backgroundPosition: 'center',
		},
	});

	const cardClasses = cardUseStyles();

	const handleCardClick = () => {
		handleClick(ancestry.id);
	};

	return (
		<Card
			onClick={handleCardClick}
			className={classes.root}
			variant={ancestry.isClicked ? 'elevation' : 'outlined'}
			css={{
				height: '500px',
			}}
		>
			{ancestry.isClicked && (
				<CheckCircleIcon
					style={{
						color: green.A200,
						backgroundColor: 'white',
						position: 'absolute',
						borderRadius: '100%',
						top: '-5px',
						right: '-5px',
						zIndex: 1,
					}}
				/>
			)}
			<CardActionArea>
				<CardMedia
					className={cardClasses.root}
					image={`/images/${ancestry.imageUrl}`}
					title={ancestry.name}
					css={{
						backgroundPosition: 'top',
					}}
				/>
				<CardContent css={{ paddingTop: '0' }}>
					<Typography gutterBottom variant="h5" component="h2" align="center">
						{formatMessage({ id: `ancestry.${ancestry.name}` })}
					</Typography>
					<Typography
						variant="body2"
						color="textSecondary"
						component="p"
						css={{ height: '100px' }}
					>
						{formatMessage({ id: `ancestry.${ancestry.name}.description` })}
					</Typography>
					<Divider variant="fullWidth" />
					<Typography color="secondary">
						{formatMessage({ id: 'bonus' })} :{' '}
						{ancestry.abilityBoost.map((anc, index) => {
							return (
								<Typography key={index} component="strong" color="secondary">
									{formatMessage({ id: `common.${anc}` })}{' '}
								</Typography>
							);
						})}
					</Typography>
					<Typography color="error">
						{formatMessage({ id: 'malus' })} :{' '}
						{ancestry.abilityFlaw.map((anc, index) => {
							return (
								<Typography key={index} component="strong" color="error">
									{formatMessage({ id: `common.${anc}` })}{' '}
								</Typography>
							);
						})}
					</Typography>
				</CardContent>
			</CardActionArea>
		</Card>
	);
};
