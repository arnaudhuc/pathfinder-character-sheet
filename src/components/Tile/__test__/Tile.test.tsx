/** @jsx jsx */
import { jsx } from '@emotion/core';
import renderer from 'react-test-renderer';
import { Tile } from '../Tile';
import { ITileInfo } from '../../../models/interface/ancestries';
import { EAbility } from '../../../models/enum/ability';
import createComponentWithIntl from '../../../utils/createComponentWithIntl';

describe('Tile', () => {
	const ancestry: ITileInfo = {
		id: 1,
		name: 'elf',
		imageUrl: 'elf.jpg',
		abilityBoost: [EAbility.DEX, EAbility.INT, EAbility.FRE],
		abilityFlaw: [EAbility.CON],
		isClicked: false,
	};
	test('should render with default props', () => {
		const container = createComponentWithIntl(
			<Tile ancestry={ancestry} handleClick={() => true} tileType="ancestry" />,
		);

		expect(container).toMatchSnapshot();
	});

	test('should render when the Tile is clicked', () => {
		ancestry.isClicked = true;
		const container = createComponentWithIntl(
			<Tile ancestry={ancestry} handleClick={() => true} tileType="ancestry" />,
		);

		expect(container).toMatchSnapshot();
	});
});
