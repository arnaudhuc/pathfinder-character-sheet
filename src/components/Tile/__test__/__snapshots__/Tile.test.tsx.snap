// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile should render when the Tile is clicked 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-72 MuiPaper-elevation1 MuiPaper-rounded"
  onClick={[Function]}
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root"
    focusable="false"
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": "100%",
        "color": "#69f0ae",
        "position": "absolute",
        "right": "-5px",
        "top": "-5px",
        "zIndex": 1,
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
    />
  </svg>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <div
      className="MuiCardMedia-root makeStyles-root-73"
      style={
        Object {
          "backgroundImage": "url(\\"https://s1.qwant.com/thumbr/0x380/0/7/1facf8942b993307b279a5bde7cd835bc9596708014bacee90d0a7639ac13e/a9a41b339e1de3975c4f5227ae28e939.png?u=https%3A%2F%2Fi.pinimg.com%2Foriginals%2Fa9%2Fa4%2F1b%2Fa9a41b339e1de3975c4f5227ae28e939.png\\")",
        }
      }
      title="Le roublard est un maître des activités douteuses qui frappe quand ses ennemies s'y attendent le moins."
    />
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom MuiTypography-alignCenter"
      >
        Roublard
      </h2>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        Le roublard est un maître des activités douteuses qui frappe quand ses ennemies s'y attendent le moins.
      </p>
    </div>
    <span
      className="MuiCardActionArea-focusHighlight"
    />
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`Tile should render with default props 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-outlined MuiPaper-rounded"
  onClick={[Function]}
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <div
      className="MuiCardMedia-root makeStyles-root-2"
      style={
        Object {
          "backgroundImage": "url(\\"https://s1.qwant.com/thumbr/0x380/0/7/1facf8942b993307b279a5bde7cd835bc9596708014bacee90d0a7639ac13e/a9a41b339e1de3975c4f5227ae28e939.png?u=https%3A%2F%2Fi.pinimg.com%2Foriginals%2Fa9%2Fa4%2F1b%2Fa9a41b339e1de3975c4f5227ae28e939.png\\")",
        }
      }
      title="Le roublard est un maître des activités douteuses qui frappe quand ses ennemies s'y attendent le moins."
    />
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom MuiTypography-alignCenter"
      >
        Roublard
      </h2>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        Le roublard est un maître des activités douteuses qui frappe quand ses ennemies s'y attendent le moins.
      </p>
    </div>
    <span
      className="MuiCardActionArea-focusHighlight"
    />
  </button>
</div>
`;
